src/prototype_room_memory.js

```diff
@@ -52,6 +59,9 @@ Room.prototype.getMemoryCostMatrix = function() {
     if (!this.memory.costMatrix || !this.memory.costMatrix.base) {
       return;
     }
+    if (this.shouldRecalcCostmatrix()) {
```

Comment:
I'm not sure if the costMatrix needs to recalculated. The costMatrix should only be based on static room data, so it doesn't matter when you calculate it.

If we should introduce costmatrixes for owned and unowned rooms, I think we should trigger a recalculate when the room in claimed.